examples/pytorch_lightning_distributed/nnclr.py

```diff
@@ -0,0 +1,74 @@
+import torch
+from torch import nn
+import torchvision
+import pytorch_lightning as pl
+
+from lightly.data import LightlyDataset
+from lightly.data import SimCLRCollateFunction
+from lightly.loss import NTXentLoss
+from lightly.models.modules import NNCLRProjectionHead
+from lightly.models.modules import NNCLRPredictionHead
+from lightly.models.modules import NNMemoryBankModule
+
+
+class NNCLR(pl.LightningModule):
+    def __init__(self):
+        super().__init__()
+        resnet = torchvision.models.resnet18()
+        self.backbone = nn.Sequential(*list(resnet.children())[:-1])
+        self.projection_head = NNCLRProjectionHead(512, 512, 128)
+        self.prediction_head = NNCLRPredictionHead(128, 512, 128)
+        self.memory_bank = NNMemoryBankModule(size=4096)
+
+        self.criterion = NTXentLoss()
```

Comment:
can we not gather here as well?

Reply:
As in moco we are using a memory bank so it doesn't make sense to gather here.